go/database/mpt/io/live_test.go

```diff
@@ -287,3 +282,64 @@ func TestCheckEmptyDirectory_FailsIfDirectoryContainsADirectory(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 }
+
+func TestIO_ExportLiveFromArchive(t *testing.T) {
+	// Create a small Archive from which we export LiveDB genesis.
+	sourceDir := t.TempDir()
+	source, err := mpt.OpenArchiveTrie(sourceDir, mpt.S5ArchiveConfig, mpt.NodeCacheConfig{Capacity: 1024})
+	if err != nil {
+		t.Fatalf("failed to create archive: %v", err)
+	}
+	_ = fillTestBlocksIntoArchive(t, source)
+	exportedBlockHeight := uint64(2)
+	hash, err := source.GetHash(exportedBlockHeight)
+	if err != nil {
+		t.Fatalf("cannot get hash: %v", err)
+	}
+
+	if err := source.Close(); err != nil {
+		t.Fatalf("failed to close source archive: %v", err)
+	}
+
+	// Export live database from archive.
+	buffer := new(bytes.Buffer)
+	if err := ExportFromArchive(context.Background(), sourceDir, buffer, exportedBlockHeight); err != nil {
+		t.Fatalf("failed to export Archive: %v", err)
+	}
+
+	// Import live database.
+	targetDir := t.TempDir()
+	if err := ImportLiveDb(targetDir, buffer); err != nil {
+		t.Fatalf("failed to import DB: %v", err)
+	}
+
+	if err := mpt.VerifyFileLiveTrie(targetDir, mpt.S5LiveConfig, nil); err != nil {
+		t.Fatalf("verification of imported DB failed: %v", err)
+	}
+
+	db, err := mpt.OpenGoFileState(targetDir, mpt.S5LiveConfig, mpt.NodeCacheConfig{Capacity: 1024})
+	if err != nil {
+		t.Fatalf("failed to open recovered DB: %v", err)
+	}
+	defer db.Close()
+
+	exists, err := db.Exists(common.Address{1})
```

Comment:
even if there are only two addresses, I would prefer to do this in a loop, the final code would be more compact

Reply:
Correct me if Im wrong but `address(1)` should be inside and `address(2)` should not be inside meaning the loop does not make sense.

Reply:
got it now. I did not fully understand this test, because it relies on data inserted in `fillTestBlocksIntoArchive`. I would suggest to create and test data in this test - have a loop that inserts N blocks, a few addresses each, export a block, and check that all addresses except the exported ones are not present in the newly imported db, and the imported ones are present.  It would be more visible what is happening. 

Other tests using data from `fillTestBlocksIntoArchive` do not directly test addresses created by this method. But since this method does, it would be more readable to create and test the data one next to another.